src/kafka/producer/Producer.test.ts

```diff
@@ -89,8 +87,17 @@ describe('Producer', () => {
       )
     })
 
+    it('should fail if send fails', async () => {
+      await producer.connect()
+      mockKafkaProducer.send.mockRejectedValueOnce(new Error('Failed to send'))
+
+      await expect(producer.produce(topic, mockMessage as any)).rejects.toThrowError(new Error('Failed to send'))
```

Comment:
The error you are catch comes from `kafkajs` producer error, exactly from [here](https://github.com/tulios/kafkajs/blob/0b70749931a1be3bfce81ca48ad2cf744e7e5e23/src/producer/messageProducer.js#L96). 
```js
if (
          error.name === 'KafkaJSConnectionError' ||
          (error.name === 'KafkaJSProtocolError' && error.retriable)
        ) {
          logger.error(`Failed to send messages: ${error.message}`, { retryCount, retryTime })
          await cluster.refreshMetadata()
          throw error
        }
```
KafkaJS implemented [error types](https://github.com/tulios/kafkajs/blob/0b70749931a1be3bfce81ca48ad2cf744e7e5e23/src/errors.js) and I believe instead of the message what seems to be a weak validation you should be checking agains error.names
```
error.name === 'KafkaJSConnectionError' || error.name === 'KafkaJSProtocolError'
```

Reply:
KafkaJS is mocked, I just want to check that the error thrown is the same as the kafkaJS error (mocked). The real implementation is not relevant.